### ISSUE TITLE
Add fallback query for blocks we don't have

### ISSUE THREAD

**Issue by drewstone** [edgeware-builders/substrate-eth-light-client#14]
We currently query an RPC (INFURA) for the latest block and attempt to fetch the proof and add the header for that block. Potentially, we miss intermediate blocks from our current state to this latest state.

We should build a mechanism that:
- Checks current state
- Looks at how far ahead the latest block is from current state
- Fetch all intermediate blocks as long as they are less than some threshold **N**.
- Fetch all proofs for these blocks and add all block headers in order.

**Comment by drewstone** [comment]
Basically here: https://github.com/edgeware-builders/substrate-eth-light-client/blob/master/pallets/worker/src/lib.rs#L775
we would add:

```
fn fetch_block_header(block_num: u64) -> Result<types::BlockHeader, http::Error> {
    // Make a post request to an eth chain
    let hexEncodedU64 = encode_function(block_num);
    let body = br#"{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":[hexEncodedU64, false],"id":1}"#;
    let request: http::Request = http::Request::post(
        "https://mainnet.infura.io/v3/b5f870422ee5454fb11937e947154cd2",
        [&body[..]].to_vec(),
    );
    ...
}
```

Here's a sample from INFURA
```
curl https://mainnet.infura.io/v3/YOUR-PROJECT-ID \
    -X POST \
    -H "Content-Type: application/json" \
    -d '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params": ["0x5BAD55",false],"id":1}'
```

**Comment by shekohex** [comment]
Solved in 0bc505cd4334e1b01673cc3669e70d40e94b8b02
